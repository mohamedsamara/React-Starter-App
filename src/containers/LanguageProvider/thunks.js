/*
 *
 * LanguageProvider thunks
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import { toastr } from 'react-redux-toastr';
import { push, goBack } from 'connected-react-router';

import { handleSetLocale } from './actions';

export function setLocale(lang) {
  return function(dispatch, getState) {
    localStorage.appLang = lang;
    return dispatch(handleSetLocale(lang));
  };
}
