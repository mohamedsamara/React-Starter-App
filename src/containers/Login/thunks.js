/*
 * 
 * Login
 * 
 */


// Do this in every file where you use `fetch`
import fetch from 'cross-fetch'
import validator from "validator";
import { toastr } from 'react-redux-toastr';
import { push, goBack } from 'connected-react-router';


import {} from './actions';


/*
export function fetch() {
  return function(dispatch, getState) {
    return fetch('', {
      method: 'GET'
    })
      .then(response => {
        return response.json();
      })
      .then(data => {
      })
      .catch(error => {});
  };
}
*/
