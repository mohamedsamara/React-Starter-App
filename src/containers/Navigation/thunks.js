/*
 *
 * Navigation
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import validator from 'validator';

import { handleToggleDrawer } from './actions';

// Toggle Drawer
export function toggleDrawer(side, open) {
  const isDrawerOpen = {};
  isDrawerOpen.side = side;
  isDrawerOpen.open = open;

  return function(dispatch, getState) {
    return dispatch(handleToggleDrawer(isDrawerOpen));
  };
}
