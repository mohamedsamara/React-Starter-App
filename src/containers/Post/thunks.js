/*
 *
 * Post thunks
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import validator from 'validator';
import { toastr } from 'react-redux-toastr';
import { push, goBack } from 'connected-react-router';

import {
  fetchPostsRequest,
  fetchPostsSuccess,
  handlePostFailure,
  deletePostSuccess,
} from './actions';

// post failure handler
export function postFailureHandler(errorData) {
  return function(dispatch, getState) {
    return dispatch(handlePostFailure(errorData));
  };
}

// fetch posts API
export function fetchPosts() {
  return function(dispatch, getState) {
    dispatch(fetchPostsRequest(true));

    return fetch('https://jsonplaceholder.typicode.com/posts', {
      method: 'GET',
    })
      .then(response => {
        if (!response.ok) {
          throw Error(response.statusText);
        }

        return response.json();
      })
      .then(data => {
        toastr.success('Success', 'Posts fetched successfully');
        dispatch(fetchPostsSuccess(data));
        dispatch(fetchPostsRequest(false));
      })
      .catch(error => {
        const errorData = {};
        errorData.isError = true;
        errorData.errorMessage = 'Failed to fetch Posts';

        toastr.error('Error', 'Error fetching todos');

        dispatch(postFailureHandler(errorData));
      });
  };
}

// delete post API
export function deletePost(id, index) {
  return function(dispatch, getState) {
    return fetch(`https://jsonplaceholder.typicode.com/posts/${id}`, {
      method: 'DELETE',
    })
      .then(response => {
        if (!response.ok) {
          throw Error(response.statusText);
        }

        return response.json();
      })
      .then(data => {
        // check if the post is deleted then dispatch an action to remove the post from the store.
        toastr.success('Success', 'Post deleted successfully');
        dispatch(deletePostSuccess(index));
      })
      .catch(error => {
        const errorData = {};
        errorData.isError = true;
        errorData.errorMessage = 'Failed to delete post';

        toastr.error('Error', 'Error deleting post');

        dispatch(postFailureHandler(errorData));
      });
  };
}
