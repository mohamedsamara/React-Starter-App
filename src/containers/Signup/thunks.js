/*
 *
 * Signup thunks
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import validator from 'validator';

import {} from './actions';

/*
export function fetch() {
  return function(dispatch, getState) {
    return fetch('', {
      method: 'GET'
    })
      .then(response => {
        return response.json();
      })
      .then(data => {
      })
      .catch(error => {});
  };
}
*/
