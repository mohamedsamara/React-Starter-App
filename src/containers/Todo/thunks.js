/*
 *
 * Todo
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import validator from 'validator';
import { toastr } from 'react-redux-toastr';
import { push, goBack } from 'connected-react-router';

import {
  fetchTodosRequest,
  fetchTodosSuccess,
  fetchTodosFailure
} from './actions';

export function fetchTodos() {
  return function(dispatch, getState) {
    dispatch(checkTodos());

    return fetch('https://jsonplaceholder.typicode.com/todos', {
      method: 'GET'
    })
      .then(response => {
        if (!response.ok) {
          throw Error(response.statusText);
        }

        return response.json();
      })
      .then(data => {
        toastr.success('Success', `Todos Fetched Successfully`);
        dispatch(fetchTodosSuccess(data));
        dispatch(fetchTodosRequest(false));
      })
      .catch(error => {
        let errorData = {};
        errorData.isError = true;
        errorData.errorMessage = 'Failed to fetch Todos';

        toastr.error('Error', `Error Fetching Todos`);

        dispatch(fetchTodosFailure(errorData));
      });
  };
}

// Check Todos
export function checkTodos() {
  return function(dispatch, getState) {
    if (getState().todo.todos.length == 0) {
      dispatch(fetchTodosRequest(true));
    }
  };
}
