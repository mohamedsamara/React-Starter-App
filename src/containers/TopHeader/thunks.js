/*
 *
 * TopHeader thunks
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import { toastr } from 'react-redux-toastr';
import { push, goBack } from 'connected-react-router';

import { handleOpenMenu, handleCloseMenu } from './actions';

// Open Menu
export function openMenu(target) {
  return function(dispatch, getState) {
    return dispatch(handleOpenMenu(target));
  };
}

// Close Menu
export function closeMenu() {
  return function(dispatch, getState) {
    return dispatch(handleCloseMenu());
  };
}
