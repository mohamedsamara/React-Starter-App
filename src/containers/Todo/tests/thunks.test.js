import { fetchTodos } from '../thunks';

describe('Todos Thunk', () => {
  it('should dispatch correct actions when fetching todos', () => {
    const mockDispatch = jest.fn();
    const result = fetchTodos();

    expect(typeof fetchTodos()).toEqual('function');

    return result(mockDispatch).then(() => {
      expect(mockDispatch.mock.calls).toMatchSnapshot();
    });
  });
});
