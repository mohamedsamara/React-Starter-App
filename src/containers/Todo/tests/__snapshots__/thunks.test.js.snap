// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos Thunk should dispatch correct actions when fetching todos 1`] = `
Array [
  Array [
    [Function],
  ],
  Array [
    Object {
      "todos": Array [
        Object {
          "completed": true,
          "id": 171,
          "title": "fugiat aut voluptatibus corrupti deleniti velit iste odio",
          "userId": 9,
        },
        Object {
          "completed": false,
          "id": 172,
          "title": "et provident amet rerum consectetur et voluptatum",
          "userId": 9,
        },
        Object {
          "completed": false,
          "id": 173,
          "title": "harum ad aperiam quis",
          "userId": 9,
        },
        Object {
          "completed": false,
          "id": 174,
          "title": "similique aut quo",
          "userId": 9,
        },
        Object {
          "completed": true,
          "id": 175,
          "title": "laudantium eius officia perferendis provident perspiciatis asperiores",
          "userId": 9,
        },
        Object {
          "completed": false,
          "id": 176,
          "title": "magni soluta corrupti ut maiores rem quidem",
          "userId": 9,
        },
        Object {
          "completed": false,
          "id": 177,
          "title": "et placeat temporibus voluptas est tempora quos quibusdam",
          "userId": 9,
        },
        Object {
          "completed": true,
          "id": 178,
          "title": "nesciunt itaque commodi tempore",
          "userId": 9,
        },
        Object {
          "completed": true,
          "id": 179,
          "title": "omnis consequuntur cupiditate impedit itaque ipsam quo",
          "userId": 9,
        },
        Object {
          "completed": true,
          "id": 180,
          "title": "debitis nisi et dolorem repellat et",
          "userId": 9,
        },
        Object {
          "completed": false,
          "id": 181,
          "title": "ut cupiditate sequi aliquam fuga maiores",
          "userId": 10,
        },
        Object {
          "completed": true,
          "id": 182,
          "title": "inventore saepe cumque et aut illum enim",
          "userId": 10,
        },
        Object {
          "completed": true,
          "id": 183,
          "title": "omnis nulla eum aliquam distinctio",
          "userId": 10,
        },
        Object {
          "completed": false,
          "id": 184,
          "title": "molestias modi perferendis perspiciatis",
          "userId": 10,
        },
        Object {
          "completed": false,
          "id": 185,
          "title": "voluptates dignissimos sed doloribus animi quaerat aut",
          "userId": 10,
        },
        Object {
          "completed": false,
          "id": 186,
          "title": "explicabo odio est et",
          "userId": 10,
        },
        Object {
          "completed": false,
          "id": 187,
          "title": "consequuntur animi possimus",
          "userId": 10,
        },
        Object {
          "completed": true,
          "id": 188,
          "title": "vel non beatae est",
          "userId": 10,
        },
        Object {
          "completed": true,
          "id": 189,
          "title": "culpa eius et voluptatem et",
          "userId": 10,
        },
        Object {
          "completed": true,
          "id": 190,
          "title": "accusamus sint iusto et voluptatem exercitationem",
          "userId": 10,
        },
        Object {
          "completed": true,
          "id": 191,
          "title": "temporibus atque distinctio omnis eius impedit tempore molestias pariatur",
          "userId": 10,
        },
        Object {
          "completed": false,
          "id": 192,
          "title": "ut quas possimus exercitationem sint voluptates",
          "userId": 10,
        },
        Object {
          "completed": true,
          "id": 193,
          "title": "rerum debitis voluptatem qui eveniet tempora distinctio a",
          "userId": 10,
        },
        Object {
          "completed": false,
          "id": 194,
          "title": "sed ut vero sit molestiae",
          "userId": 10,
        },
        Object {
          "completed": true,
          "id": 195,
          "title": "rerum ex veniam mollitia voluptatibus pariatur",
          "userId": 10,
        },
        Object {
          "completed": true,
          "id": 196,
          "title": "consequuntur aut ut fugit similique",
          "userId": 10,
        },
        Object {
          "completed": true,
          "id": 197,
          "title": "dignissimos quo nobis earum saepe",
          "userId": 10,
        },
        Object {
          "completed": true,
          "id": 198,
          "title": "quis eius est sint explicabo",
          "userId": 10,
        },
        Object {
          "completed": true,
          "id": 199,
          "title": "numquam repellendus a magnam",
          "userId": 10,
        },
        Object {
          "completed": false,
          "id": 200,
          "title": "ipsam aperiam voluptates qui",
          "userId": 10,
        },
      ],
      "type": "src/Todo/FETCH_TODOS_SUCCESS",
    },
  ],
  Array [
    Object {
      "isFetching": false,
      "type": "src/Todo/FETCH_TODOS_REQUEST",
    },
  ],
]
`;
